pymodbus/repl/server/main.py

```diff
@@ -138,7 +138,7 @@ def run(
         autocompletion=framers,
         help="Modbus framer to use",
     ),
-    modbus_port: str = typer.Option("5020", "--modbus-port", "-p", help="Modbus port"),
+    modbus_port: int = typer.Option(5020, "--modbus-port", "-p", help="Modbus port"),
```

Comment:
This needs a conversion to int, otherwise we receive a str.

Reply:
`typer` does the conversion already. https://typer.tiangolo.com/tutorial/parameter-types/#data-conversion

Also, since `typer` was new to me I tested it (by adding `print(modbus_port, type(modbus_port)`)

```
❯  pymodbus.server run --modbus-server tcp --modbus-port 1234

1234 <class 'int'>
```